### DISCUSSION TITLE
Trouble with go get

### DISCUSSION THREAD

**Discussion by data-diego** [pocketbase/pocketbase#1236] [Q&A] [answered]
Hi! Sorry to bother with this rather simple question but I can't seem to find any solution.

In go version `>1.17`  the `go get` command was deprecated and we have to use `go install` instead [https://go.dev/doc/go-get-install-deprecation](https://go.dev/doc/go-get-install-deprecation) this is because `go get` updates dependencies and also install commands so the go team decided to separate the functions.

Trying to run `go get -u github.com/pocketbase/pocketbase` throws an error and asks us to run it with `go install`

But when running 

`go install github.com/pocketbase/pocketbase@latest` 

the answer is 

`package github.com/pocketbase/pocketbase is not a main package`

----

Then I tried `go mod init github.com/pocketbase/pocketbase` and `go mod tidy` but at both `go build` or `go run main.go` the answer is 

`imports github.com/pocketbase/pocketbase: import cycle not allowed`

Does anyone know what I'm doing wrong?

ps. thank you for this awesome repo

**Comment by ganigeorgiev** [comment, marked as answer]
PocketBase is distributed as a Go library/package, aka. you have to embed it inside your Go program. Or in other words, create a Go file and import it like this:

```go
// main.go
package main

import (
    "log"

    "github.com/pocketbase/pocketbase"
)

func main() {
    app := pocketbase.New()

    // customize it with event hooks https://pocketbase.io/docs/event-hooks/ ...

    if err := app.Start(); err != nil {
        log.Fatal(err)
    }
}
```

Run `go mod init` (if you haven't already) and `go mod tidy` to fetch your dependencies. Then you can start your program as usual: `go run main.go serve` or `go build` and then `./yourapp serve`.

If you just want to use the prebuilt executable, you can download them from the "Assets" section that comes with each release in  https://github.com/pocketbase/pocketbase/releases. The prebuilt executable uses this main.go file - https://github.com/pocketbase/pocketbase/blob/master/examples/base/main.go

**Reply by data-diego** [reply]
Thank you! my problem was at `go mod init`, I needed to give the module name because it wasn't taking it automatically so with 

`go mod init main` and `go mod tidy` now I'm all set
